plugins/out_s3/s3.c

```diff
@@ -850,6 +842,17 @@ static int cb_s3_init(struct flb_output_instance *ins,
         }
     }
 
+    if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
+        if(ctx->upload_chunk_size > MAX_CHUNKED_UPLOAD_COMPRESS_SIZE) {
+            flb_plg_error(ctx->ins, "upload_chunk_size in compress multipart upload cannot over 5GB");
```

Comment:
how do we decide it is put object mode or multi upload mode?

Reply:
I do not need to check whether it is put object mode or multi upload mode, because put object mode limitation is 50MB and it is checked already. So we do not have max limitation for multi upload mode yet